creator-node/src/middlewares.js

```diff
@@ -481,8 +481,30 @@ async function ensureValidSPMiddleware(req, res, next) {
   next()
 }
 
+// General middleware for routes to initiate metric tracking
```

Comment:
makes sense why you did this, thinking through implications
- we already have prometheus middleware which gives us simple route metrics, i'm assuming that does not conflict with this at all
- this assumes each route will have at most 1 custom defined metric - thats ok for now, but something to think about